engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -137,25 +133,18 @@
 	private final HibernateConstraintValidatorInitializationContext constraintValidatorInitializationContext;
 
 	public ValidatorImpl(ConstraintValidatorFactory constraintValidatorFactory,
-			MessageInterpolator messageInterpolator,
-			TraversableResolver traversableResolver,
 			BeanMetaDataManager beanMetaDataManager,
-			ExecutableParameterNameProvider parameterNameProvider,
-			ClockProvider clockProvider,
-			ScriptEvaluatorFactory scriptEvaluatorFactory,
 			ValueExtractorManager valueExtractorManager,
 			ConstraintValidatorManager constraintValidatorManager,
 			ValidationOrderGenerator validationOrderGenerator,
-			boolean failFast,
-			boolean traversableResolverResultCacheEnabled,
-			Duration temporalValidationTolerance) {
+			ValidatorFactoryContext validatorFactoryContext) {
 		this.constraintValidatorFactory = constraintValidatorFactory;
 		this.beanMetaDataManager = beanMetaDataManager;
 		this.valueExtractorManager = valueExtractorManager;
 		this.constraintValidatorManager = constraintValidatorManager;
 		this.validationOrderGenerator = validationOrderGenerator;
-		this.traversableResolver = traversableResolver;
-		this.validatorScopedContext = new ValidatorScopedContext( messageInterpolator, parameterNameProvider, clockProvider, temporalValidationTolerance, scriptEvaluatorFactory, failFast, traversableResolverResultCacheEnabled );
+		this.validatorScopedContext = validatorFactoryContext.toValidatorScopedContext();
```

Comment:
I think I would prefer the opposite i.e. having a `ValidatorScopedContext` constructor taking a validator factory context. We might want to add other things in it at one point.